docs/deprecation.md

```diff
@@ -0,0 +1,23 @@
+# Deprecation Policy
+
+## User-facing commands and their arguments
+
+Whenever you rename a command or make a previously optional argument required,
+we require that you preserve the old command invocations keeps on working for 6
+months (so 6 releases, since we release monthly) with a deprecation message.
+The message should inform the user that the previous workflow is deprecated
+and to be removed in the future.
+
+## Packaging commands and niche commands
+
+For commands with a niche user audience or something we assume is rarely used
+(we sadly have no data), we take the liberity to remove the old behavior within
+two releases. This means that you can change the old command to immediately
+error.
+
+## Third-Party dependencies
+
+For third-party dependencies which previously were used for a core functionality
```

Comment:
(For discussion) This section doesn't make a lot of sense to me. Perhaps it's OK to just remove it, in favor of a possible future "packager-facing deprecations" section.

Reply:
I agree. I think there are two different things involved in this transition. One is the user-visible changes. That's what I think we need to be careful with. That's not specific to third-party dependencies. The other thing is the actual removal of the dependency. I don't think we should have a deprecation policy for that (we should remove dependencies as soon as they're unused).